azure/services/groups/spec.go

```diff
@@ -60,7 +61,11 @@ func (s *GroupSpec) Parameters(ctx context.Context, existing *asoresourcesv1.Res
 				Additional:  s.AdditionalTags,
 			}),
 		},
-	}, nil
+	}
+	if s.AzureName != "" {
```

Comment:
I'd vote to also set `AzureName` for the main resource group instead of requiring ourselves to handle the case where it might not be set.

Reply:
ASO seems to be populating AzureName when AzureName is left empty. My fix is more of a hot patch on the logic flow where AzureName differs. 

Are you suggesting to set AzureName at Scope level?

Reply:
Thinking about this, ClusterScope also needs to be updated with these changes.

Reply:
Yeah, I suppose we could leave this undefined in the scopes where we want it the same anyway and remove this check then to let ASO default it.

Reply:
I am not sure I understand.
Are you suggesting to remove this `if s.AzureName != "" {` from `groups/spec.go`  ? 
If yes, how will CAPZ propagate the resourceGroup.AzureNames that we capture at Cluster and ManagedControlPlane Scopes to ASO CRs?